src/plots/ternary/ternary.js

```diff
@@ -430,6 +457,11 @@ proto.drawAxes = function(doTitles) {
     }
 };
 
+function strTickLayout(axLayout) {
+    return axLayout.ticks + String(axLayout.ticklen) + String(axLayout.showticklabels);
```

Comment:
`Axes.doTicks` is able to re-style existing nodes (e.g. relayout calls involving `tickfont`) e.g. see this test:

https://github.com/plotly/plotly.js/blob/2a667d03f1045b75c5c568bd30e33f28df36b9d5/test/jasmine/tests/ternary_test.js#L352-L383

but it can't handle hide/show. Cartesian axes rely on this block:

https://github.com/plotly/plotly.js/blob/2a667d03f1045b75c5c568bd30e33f28df36b9d5/src/plots/cartesian/axes.js#L1550-L1555

which isn't compatible with non-cartesian subplots.